lib/ansible/modules/find.py

```diff
@@ -352,14 +358,21 @@ def contentfilter(fsname, pattern, read_whole_file=False):
     prog = re.compile(pattern)
 
     try:
-        with open(fsname) as f:
+        with open(fsname, encoding=encoding) as f:
             if read_whole_file:
                 return bool(prog.search(f.read()))
 
             for line in f:
                 if prog.match(line):
                     return True
 
+    except LookupError as e:
+        raise e
+    except UnicodeDecodeError as e:
+        if encoding is None:
+            encoding = 'None (default determined by the Python built-in function "open")'
+        msg = f'Failed to read the file {fsname} due to an encoding error. current encoding: {encoding}'
+        raise Exception(msg) from e
     except Exception:
         pass
```

Comment:
Is this needed? After this PR, these lines are not covered anymore. Besides, the code above turns some errors into `Exception` anyway. If yes — can this be covered with tests?

Reply:
https://app.codecov.io/gh/ansible/ansible/commit/aa40167f4031eab9b0df785e0ef3fac25b1d184e#b3f813d4fc64d8e0ed291de8bfa23e37-R377

Reply:
It appears that no testing was conducted on it before this pull request. The code may be removed if it is considered dead code.

Reply:
@webknjaz It seemed necessary for backwards compatibility to me https://github.com/ansible/ansible/pull/82284#discussion_r1406334179. There are layers of exception handling in this module (your favorite, I know). Now that `encoding` can be configured it makes sense to let those specific errors related to encoding bubble up to the module result. Not sure how there could have been test coverage since refactoring the method in backwards incompatible ways didn't cause any test failures...